### PR TITLE
[Datasets] [Hotfix] Add ray_remote_args to read_text.

### PR DESCRIPTION
**Pull request by clarkzinzow**
Adds `ray_remote_args` to `ray.data.read_text()`. This exists on all other `ray.data.read_*` APIs, so it appears that `read_text()` was missed.

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [x] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

File: python/ray/data/tests/test_dataset_formats.py [hunk 1]
```diff
@@ -1444,6 +1492,44 @@ def test_csv_write_block_path_provider(
...
+    )
+    assert sorted(ds.take()) == ["goodbye", "hello", "world"]
+
+    blocks = ds.get_internal_block_refs()
+    ray.wait(blocks, num_returns=len(blocks), fetch_local=False)
+    location_data = ray.experimental.get_object_locations(blocks)
+    locations = []
+    for block in blocks:
+        locations.extend(location_data[block]["node_ids"])
+    assert set(locations) == {bar_node_id}, locations
```
Comment (clarkzinzow) [inline python/ray/data/tests/test_dataset_formats.py:1530]: This straightforward test was failing locally for me, not sure why. 🤔

**Review by jjyao — APPROVED** [review]

**Comment by clarkzinzow** [comment]
Will be coming back to this in the next week or so.

**Comment by krfricke** [comment]
I've taken a look.

- The test is stochastic, as sometimes the remote tasks are scheduled only on the `bar` node
- There were two sources of errors
  1. We didn't pass the remote args to `flat_map()`, which were sometimes scheduled on the `foo` node
  2. `ds.take()` also schedules remote tasks, which were sometimes scheduled on the `foo` node
- To resolve this, we're now passing remote args to `flat_map`()
- I don't think we need to solve `ds.take()` - but we need to move that assertion to the bottom of the test.

This all passes locally for me every time now.

**Review by krfricke — APPROVED** [review]
Bit of a self-accept but still - cc @clarkzinzow to merge

File: python/ray/data/read_api.py [hunk 2]
```diff
@@ -579,6 +580,7 @@ def read_text(
         filesystem: The filesystem implementation to read from.
         parallelism: The requested parallelism of the read. Parallelism may be
             limited by the number of files of the dataset.
+        ray_remote_args: Kwargs passed to ray.remote in the read tasks.
```
Comment (clarkzinzow) [inline python/ray/data/read_api.py:583]:
```suggestion
        ray_remote_args: Kwargs passed to ray.remote in the read tasks and
            in the subsequent text decoding map task.
```

**Review by clarkzinzow — COMMENTED** [review]
@krfricke Thanks a million for finding the blocker bug! Definitely a 🤦  moment for me. 😁 

I think this is good to merge, with a P2 follow-up around adding a `TextDatasource` to get rid of this two-stage read.

**Pull request merged**